genesyscloud/provider/provider.go

```diff
@@ -178,24 +179,9 @@ type ProviderMeta struct {
 
 func configure(version string) schema.ConfigureContextFunc {
 	return func(context context.Context, data *schema.ResourceData) (interface{}, diag.Diagnostics) {
-		// Initialize a single client if we have an access token
```

Comment:
What is this?

Reply:
since we have a single client for access token, we are seeing issues when getResourceFunc context is called more than once anywhere in the already existing context.  This is forming a race condition where both the getResourceContexts are waiting for a config pool and we only have one, therefore making the export hung for infinite time.

this issue happens when a access token is provided instead of credential. and the exporter gets hung at the filter logic of ruleset ( using exporter instance of skillset to get to getllroutingskills)

root cause for both the issues
https://inindca.atlassian.net/browse/DEVTOOLING-750
https://inindca.atlassian.net/browse/DEVTOOLING-642

Reply:
Great find!